lib/Doctrine/ORM/Cache/DefaultEntityHydrator.php

```diff
@@ -73,7 +73,7 @@ public function __construct(EntityManagerInterface $em)
     public function buildCacheEntry(ClassMetadata $metadata, EntityCacheKey $key, $entity)
     {
         $data = $this->uow->getOriginalEntityData($entity);
-        $data = array_merge($data, $key->identifier); // why update has no identifier values ?
```

Comment:
I'm not really sure why is the line even there, I did not find the case when identifier would not be in the original entity data. But this is the line, that causes the described issue.

And in general, why should the cached entity data contain data from cache key...